bin/build_update_subfolders.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+
+import send_email
+import synapseclient
+import synapseutils
+import sys
+
+from typing import Any, List, Tuple, Union
+
+
+def is_new_submitter(syn, submitter_id, parent_folder_id):
+    """
+    Used to check if the submitter is new or already has an existing folder.
+
+    This function uses ``synapseutils.walk`` to crawl through the level 1 subfolders
+    (i.e. The folders 1 level below the Parent-Folder, that are named after the submitter's synapse ID) to
+    verify whether or not a submitter already has an existing folder created.
+    """
+
+    submitter_subfolders = []
+    # Setting ``include_types`` to only Folders so that Files are excluded, and walk is faster.
+    for _, level1_subfolders, _ in synapseutils.walk(syn=syn, synId=parent_folder_id,
+                                                     includeTypes=["folder"]
+                                                     ):
+        for folder_name, synid in level1_subfolders:
+            submitter_subfolders.append(folder_name)
+
+    if submitter_id in submitter_subfolders:
+        return False
+
+    return True
+
+
+def build_subfolder(syn: synapseclient.Synapse, folder_name: str, parent_folder: Union[str, synapseclient.Entity]) -> synapseclient.Entity:
+    """
+    Builds a subfolder under the designated ``parent_folder``.
+
+    Arguments:
+        syn: A Synapse Python client instance
+        folder_name: The name of the subfolder to be created
+        parent_folder: A synapse Id or Entity of the parent folder under which the subfolder will live
+
+    Returns:
+        The created Synapse Folder entity
+    """
+
+    # Create Folder object
+    subfolder = synapseclient.Folder(
+        name=folder_name, parent=parent_folder
+    )
+    subfolder = syn.store(obj=subfolder)
+
+    return subfolder
+
+
+# TODO: https://sagebionetworks.jira.com/browse/IBCDPE-809
+# def update_subfolders():
+#     """"""
+
+
+def update_permissions(syn: synapseclient.Synapse, subfolder: Union[str, synapseclient.Entity],
+                       project_folder_id: str, access_type: Any = ["CREATE", "READ", "DOWNLOAD", "UPDATE", "DELETE"]):
+    """
+    Updates the permissions (local share settings) of the given Folder/File to grant the creators of the Project
+    unique permissions.
+
+    Arguments:
+        syn: A Synapse Python client instance
+        subfolder: The Folder whose permissions will be updated
+        project_folder_id: The Project Synapse ID
+        access_type: Type of permission to be granted
+    """
+    organizers_id = syn.get(project_folder_id, downloadFile=False).createdBy
+    syn.setPermissions(
+        subfolder, principalId=organizers_id, accessType=access_type
+        )
+
+
+def get_parent_and_project_folder_id(syn: synapseclient.Synapse, parent_folder: str, project_name: str) -> Tuple[str, str]:
```

Comment:
Nit: 

1.  I understand your rationale around adding this in within a function, but does it add complexity by returning a tuple?

    ```
    project_id = syn.findEntityId(name=project_name)
    parent_folder_id = syn.findEntityId(name=parent_folder, parent=project_id)
    ```

2. Look at `namedtuple` so that we aren't relying on positions

Reply:
The tuple does add complexity. I ended up trying out NamedTuples and it seems to be working fine based off my latest tower run. However, I suppose this function is small enough in scope that it can be scrapped and the `syn.findEntityId` calls for Project and parent folder can be moved to the main function. I'll revisit this tomorrow.

Reply:
I think I'm going to scrap the use of NamedTuple and move these calls to `findEntityId` into the main function. The scope of this function is small enough, and the `SynapseIds(NamedTuple)` class doesn't get called enough throughout the script for it to exist in my opinion. If there are no strong objections, I'll move forward with this restructuring.